### PR TITLE
Release candidate v0.27.0

### PR DESCRIPTION
**Pull request by mlxd**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:** Release candidate builder PR for v0.27.0

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: tests/test_apply.py [hunk 1]
```diff
@@ -462,7 +462,8 @@ def test_apply_errors_qubit_state_vector(self, qubit_device_2_wires):
 
         with pytest.raises(
             ValueError,
-            match=r"State vector must have shape \(2\*\*wires,\) or \(batch_size, 2\*\*wires\).",
+            match=r"cannot reshape array of size 5 into shape \(2,2\)"
+            # match=r"State vector must have shape \(2\*\*wires,\) or \(batch_size, 2\*\*wires\).",
```
Comment (AmintorDusko) [inline tests/test_apply.py:466]: Do we need to leave this here for now?

Comment (mlxd) [reply tests/test_apply.py:466]: Nope, should've caught it. Thanks.

File: tests/test_hamiltonian_sparse.py [hunk 2]
```diff
@@ -59,7 +59,7 @@ def test_hamiltionan_expectation(self, qubit_device_3_wires, tol):
 
         dev.syncH2D()
         Hmat = qml.utils.sparse_hamiltonian(H)
-        H_sparse = qml.SparseHamiltonian(Hmat, wires=3)
+        H_sparse = qml.SparseHamiltonian(Hmat, wires=range(3))
```
Comment (AmintorDusko) [inline tests/test_hamiltonian_sparse.py:62]: Good catch, here!
Could you please go to the test definitions and change:
`def test_hamiltionan_expectation(self, qubit_device_3_wires, tol):`
to
`def test_hamiltonian_expectation(self, qubit_device_3_wires, tol):`?
I can't reach there to comment.

Comment (mlxd) [reply tests/test_hamiltonian_sparse.py:62]: Thanks Amintor, and done.

**Review by maliasadi — APPROVED** [review]
LGTM. Thanks @mlxd :tada:

**Pull request merged**